service-providers/network-requester/examples/query.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use nym_sdk::mixnet::{IncludedSurbs, MixnetClient, Recipient, ReconstructedMessage};
+use nym_service_providers_common::interface::{
+    ProviderInterfaceVersion, Request, Response, ResponseContent,
+};
+use nym_socks5_requests::{QueryRequest, Socks5Request, Socks5Response};
+
+fn parse_response(received: Vec<ReconstructedMessage>) -> Socks5Response {
+    assert_eq!(received.len(), 1);
+    let response: Response<Socks5Request> = Response::try_from_bytes(&received[0].message).unwrap();
+    match response.content {
+        ResponseContent::Control(control) => panic!("unexpected control response: {:?}", control),
+        ResponseContent::ProviderData(data) => data,
+    }
+}
+
+async fn wait_for_response(client: &mut MixnetClient) -> Socks5Response {
+    loop {
+        let next = client.wait_for_messages().await.unwrap();
+        if !next.is_empty() {
+            return parse_response(next);
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    //nym_bin_common::logging::setup_logging();
+    let mut client = MixnetClient::connect_new().await.unwrap();
+    //let provider: Recipient = "AN8eLxYWFitCkMn92zim3PrPszxJZDYyFFKP7qnnAAew.8UAxL3LwQBis6WpM3GGXaqKGaVdnLCpGJWumHT6KNdTH@77TSuVU8d1oXKbPzjec2xh4i3Wj5WwUyy9Lr36sm8gZm".parse().unwrap();
+    let provider: Recipient = "85xZUGsMi8koFGXnoKQpn3wkcajSghu99uW3P4aQqP3E.D5PdTNtPZnek28BsctU1pYGFzMSTx9z7g9EL9ryjDB5Z@9xJM74FwwHhEKKJHihD21QSZnHM2QBRMoFx9Wst6qNBS".parse().unwrap();
```

Comment:
I wonder if examples should have placeholder provider addresses for users to enter it manually because there's no guarantee what you entered here is going to be valid in a week/month, etc.

Reply:
Yep idk what's best, it's kinda assumed that the example runner will need to lookup a network-requester manually